pandas/core/groupby/groupby.py

```diff
@@ -2855,7 +2859,9 @@ def _value_counts(
             result_series.name = name
             result_series.index = index.set_names(range(len(columns)))
             result_frame = result_series.reset_index()
-            result_frame.columns = columns + [name]
+            result_frame.columns = Index(
+                columns + [name], dtype=self.grouper.groupings[0].obj.columns.dtype
```

Comment:
@rhshadrach is there a better way of getting the dtype of the original column index?

Reply:
self.obj.columns.dtype I think.

Reply:
That's a Series in SeriesGroupBy unfortunately